src/test/java/core/basesyntax/strategy/impl/PurchaseImplTest.java

```diff
@@ -0,0 +1,51 @@
+package core.basesyntax.strategy.impl;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.FruitTransaction;
+import java.util.Map;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class PurchaseImplTest {
+    private PurchaseImpl purchase;
+    private FruitTransaction fruitTransaction;
+
+    @BeforeEach
+    void setUp() {
+        purchase = new PurchaseImpl();
+        fruitTransaction = new FruitTransaction(FruitTransaction.Operation.PURCHASE, "banana", 15);
+    }
+
+    @Test
+    public void purchaseHandler_NegativeQuantity_notOk() {
+        Storage storage = new Storage();
```

Comment:
Why we create a `new Storage()` in every test case? 
You have static methods put() and get() in the Storage class.
Please fix all similar cases in all test classes.